editor/editor_interface.cpp

```diff
@@ -512,6 +513,38 @@ void EditorInterface::popup_quick_open(const Callable &p_callback, const TypedAr
 	quick_open->popup_dialog(base_types, callable_mp(this, &EditorInterface::_quick_open).bind(p_callback));
 }
 
+void EditorInterface::popup_create_dialog(const Callable &p_callback, const StringName &p_base_type, const TypedArray<StringName> &p_custom_type_blacklist, const Dictionary &p_custom_suffix) {
+	if (!create_dialog) {
+		create_dialog = memnew(CreateDialog);
+		get_base_control()->add_child(create_dialog);
+	}
+
+	HashSet<StringName> blacklist;
+	for (int i = 0; i < p_custom_type_blacklist.size(); i++) {
+		blacklist.insert(p_custom_type_blacklist[i]);
+	}
+	create_dialog->set_type_blacklist(blacklist);
+
+	HashMap<StringName, String> suffix_map;
+	for (int j = 0; j < p_custom_suffix.size(); j++) {
+		StringName k = p_custom_suffix.get_key_at_index(j);
```

Comment:
Isn't there a more efficient way? .-.
Iterating Dictionary like this is super slow. Probably won't matter in this context, but still.

Reply:
There's using `get_key_list`, standard methodology:
```cpp
List<Variant> keys;
p_custom_suffix.get_key_list(&keys);
for (StringName k : keys) {
   ...
}
```

Reply:
The `get_key_at_index` method here is `O(n^2)` btw, so good to avoid